src/commands/test/runner.py

```diff
@@ -150,22 +166,45 @@ async def invoke_test_function(self, function_name: str):
         )
 
         func = getattr(self.test_contract, function_name)
+        is_failure_expected = (
+            function_name.startswith("test_fail_")
+            if self._is_test_fail_enabled
+            else False
+        )
         # TODO: Improve stacktrace
         try:
-            call_result = await func().invoke()
-            if self._is_test_error_expected:
-                raise MissingExceptReportedException(
-                    "Expected a transaction to be reverted"
+            try:
+                call_result = await func().invoke()
+                if self._expected_error is not None:
+                    raise MissingExceptReportedException(
+                        "Expected a transaction to be reverted"
+                    )
```

Comment:
I think we should provide information here about expected error and provide proper `__str__` for the class as it's being used by the reporter to print data.

Reply:
I added info about the expected error there.

This exception is used in various places. The message changes, thereby the way how an expected error should be interpolated into the error message.